File: js/formrectest.js
const { FormRecognizerClient, AzureKeyCredential } = require("@azure/ai-form-recognizer");

async function main() {
  const endpoint = process.env["FORM_RECOGNIZER_ENDPOINT"] || "<cognitive services endpoint>";
  const apiKey = process.env["FORM_RECOGNIZER_API_KEY"] || "<api key>";
  const modelId = process.env["UNLABELED_CUSTOM_MODEL_ID"] || "<unlabeled custom model id>";
  const escapedUrl = process.env["ESCAPED_URL"] || "<escaped url>";
  const unescapedUrl = process.env["UNESCAPED_URL"] || "<unescaped url>";

  const client = new FormRecognizerClient(endpoint, new AzureKeyCredential(apiKey));

  await test(client, modelId, escapedUrl);
  await test(client, modelId, unescapedUrl);
}

async function test(client, modelId, url) {
  try {
    const poller = await client.beginRecognizeCustomFormsFromUrl(modelId, url, {
      onProgress: (state) => {
        console.log(`status: ${state.status}`);
      }
    });
    await poller.pollUntilDone();
    const response = poller.getResult();
  
    if (!response) {
      throw new Error("Expecting valid response!");
    }
  
    console.log(response.status);
  
    console.log("Errors:");
    console.log(response.errors);
  }
  catch (err) {
    console.error(err);
  }
}

main().catch((err) => {
  console.error("The sample encountered an error:", err);
});

File: python/formrectest.py
import os

class RecognizeCustomForms(object):

    endpoint = os.environ["AZURE_FORM_RECOGNIZER_ENDPOINT"]
    key = os.environ["AZURE_FORM_RECOGNIZER_KEY"]
    model_id = os.environ["CUSTOM_TRAINED_MODEL_ID"]

    escaped_url = os.environ["ESCAPED_URL"]
    unescaped_url = os.environ["UNESCAPED_URL"]

    def recognize_custom_forms(self):
        # [START recognize_custom_forms]
        from azure.core.credentials import AzureKeyCredential
        from azure.ai.formrecognizer import FormRecognizerClient
        form_recognizer_client = FormRecognizerClient(
            endpoint=self.endpoint, credential=AzureKeyCredential(self.key)
        )
        self.test(form_recognizer_client, self.escaped_url)
        self.test(form_recognizer_client, self.unescaped_url)

    def test(self, client, url):
        try:
            poller = client.begin_recognize_custom_forms_from_url(model_id=self.model_id, url=url)
            forms = poller.result()
            print(len(forms))
        except Exception as err:
            print(err)


if __name__ == '__main__':
    sample = RecognizeCustomForms()
    sample.recognize_custom_forms()